code/game/objects/items/toy_mechs.dm

```diff
@@ -11,14 +11,19 @@
 /// Max length of a mech battle
 #define MAX_BATTLE_LENGTH 50
 
-/obj/item/toy/prize
+/// Amount of mech toys
+#define ALL_MECH_TOYS 18
+
+/obj/item/toy/mecha
 	icon = 'icons/obj/toy.dmi'
-	icon_state = "ripleytoy"
+	icon_state = "fivestarstoy"
 	verb_say = "beeps"
 	verb_ask = "beeps"
 	verb_exclaim = "beeps"
 	verb_yell = "beeps"
 	w_class = WEIGHT_CLASS_SMALL
+	/// Number of the toy, used in it's desc
+	var/series_number = 0
```

Comment:
If the series numbers actual value is unimportant, would it be possible to do something like.

(pseduocode)
```
mech_types = subtypesof(/obj/item/toy/mecha)
series_number = mech_types.IndexOf(src.type)
```

Then you wouldn't need to specify it in each type, or have the ALL_MECH_TOYS define?

Reply:
i tried to number and order it in a specific way and i dont know if that would work with subtypesof

Reply:
Ah, if the numbering is important, then keep it.

Reply:
i will pr the element that does it